speechbrain/core.py

```diff
@@ -867,8 +868,17 @@ class being passed at initialization that takes only a list
 
         Override this class if there are multiple optimizers.
         """
+
+        # We remove biases and normalization parameters from weight_decay
+        all_params = self.modules.parameters()
+        if hasattr(self, "hparams"):
+            if hasattr(self.hparams, "weight_decay"):
```

Comment:
This behaviour is a little too magical in my opinion. We don't rely on a specific key in hparams anywhere. Instead, this should probably be controlled by a run_opts flag.

Reply:
I will think about it. Applying weight decay to LN and BN plainly is wrong. Simply purely wrong. Hence, no magic in forcing things to be right. And the problem here is that I need the weight_decay to be set in the hparams to access its value... a run_opts would not give me the value (no way that we put the weight decay value as a run_opt given that this is linked to the optimiser)

Reply:
You are write, however, in the sense that if someone use weight decay in the optimiser but not a hparams.weight_decay params, then it won't be triggered. I think I will instead directly check the optimiser.

Reply:
@Gastron have a look at the latest version I added your comments (check name as well and remove the magic)